---
title: "psmcr"
author: "meg hilbert"
date: "2025-04-11"
output: html_document
---

```{r}


#!/bin/bash
#SBATCH --job-name=psmcr
#SBATCH --output=psmcr.out
#SBATCH --error=psmcr.err
#SBATCH --time=0-06:00:00
#SBATCH --mail-user=hilbert.m@northeastern.edu
#SBATCH --ntasks=1
#SBATCH --ntasks-per-node=1
#SBATCH --mail-type=ALL
#SBATCH --mem=40G
#SBATCH --cpus-per-task=8
 
module load R

cd/home/hilbert.m/crms/psmcr

VCF_FILE="/courses/EEMB5538.202530/data/Zm_TomBod_MAF01MM85INDM30_AllChr.recode.vcf.gz"
REF_GENOME="/courses/EEMB5538.202530/data/Zmarina_668_v2.0.fa"
OUTPUT_DNABIN="output_dnabin.rds"
R_SCRIPT="run_vcf2dnabin.R"

cat > $R_SCRIPT << EOF

library(psmcr)

#convert VCF to Fasta files
dnabin_obj <- VCF2DNAbin("$VCF_FILE",
                         refgenome="$REF_GENOME",
                         individual= 169)

#save DNAbin project
saveRDS(dnabin_obj, file="$OUTPUT_DNABIN")

# run psmcr
psmcr_result <- psmc(dnabin_obj, niters=5, B=30, mc.cores=8)

# save psmc results
saveRDS(psmcr_result, files="psmcr.rds")
EOF

# run the r script
Rscript $R_SCRIPT


```

